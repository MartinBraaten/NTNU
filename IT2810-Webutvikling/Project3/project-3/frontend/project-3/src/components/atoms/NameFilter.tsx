//Filter for filtering on champion name
import {
    IconButton,
    InputAdornment,
    TextField,
    ThemeProvider,
} from "@mui/material";
import Box from "@mui/material/Box";
import { fieldTheme } from "../themes/themes";
import SearchIcon from "@mui/icons-material/Search";

export interface IFilter {
    change: boolean | undefined;
    setChange: (change: boolean) => void;
}

//Component for searching for specific champions by name
export default function NameFilter({ change, setChange }: IFilter) {
    return (
        <Box
            sx={{
                width: "calc(100vw - 315px)",
                float: "right",
                textAlign: "center",
            }}
            role="NameFilter"
        >
            <ThemeProvider theme={fieldTheme}>
                <TextField
                    id="outlined-basic"
                    label="Champion name"
                    variant="outlined"
                    onChange={() => {
                        setChange(!change);
                    }}
                    sx={{ width: "50vw" }}
                    InputProps={{
                        endAdornment: (
                            <InputAdornment position="end">
                                <IconButton edge="end">
                                    <SearchIcon
                                        sx={{ color: "#CFB53B" }}
                                    ></SearchIcon>
                                </IconButton>
                            </InputAdornment>
                        ),
                    }}
                />
            </ThemeProvider>
        </Box>
    );
}
