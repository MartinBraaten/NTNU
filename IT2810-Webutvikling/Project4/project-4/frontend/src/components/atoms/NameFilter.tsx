//Filter for filtering on champion name
import {
    IconButton,
    InputAdornment,
    TextField,
    ThemeProvider,
} from "@mui/material";
import Box from "@mui/material/Box";
import { fieldTheme } from "../themes/themes";
import SearchIcon from "@mui/icons-material/Search";
import { useDispatch, useSelector } from "react-redux";
import { selectFilters, setFilter } from "../store/FiltersReducer";

//Component for searching for specific champions by name
export default function NameFilter() {
    const dispatch = useDispatch();
    const filters = useSelector(selectFilters);

    return (
        <Box
            sx={{
                width: "calc(100vw - 315px)",
                float: "right",
                textAlign: "center",
            }}
            role="NameFilter"
        >
            <ThemeProvider theme={fieldTheme}>
                <TextField
                    id="outlined-basic"
                    label="Champion name"
                    variant="outlined"

                    value={filters.name}
                    onChange={(event: any) => {
                        dispatch(setFilter({ field: "name", value: event.target.value }));
                    }}

                    sx={{ width: "50vw" }}
                    InputProps={{
                        /* style: {backgroundColor:"white" ,}, */
                        endAdornment: (
                            <InputAdornment position="end">
                                <IconButton edge="end">
                                    <SearchIcon
                                        sx={{ color: "#CFB53B" }}
                                    ></SearchIcon>
                                </IconButton>
                            </InputAdornment>
                        ),
                    }}
                />
            </ThemeProvider>
        </Box>
    );
}
