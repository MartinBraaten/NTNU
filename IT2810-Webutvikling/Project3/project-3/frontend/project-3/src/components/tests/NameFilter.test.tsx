/* eslint-disable testing-library/await-async-query */
/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { render } from "@testing-library/react";
import NameFilter from "../atoms/NameFilter";
// eslint-disable-next-line testing-library/no-dom-import
import { screen } from "@testing-library/dom";

test("Renders correctly", () => {
    let { container } = render(<NameFilter change={undefined} setChange={function (change: boolean): void {
        throw new Error("Function not implemented.");
    } } />);
    expect(container).toMatchSnapshot();
});

test("Updates value on writing", () => {
    const { container } = render(
        <NameFilter change={undefined} setChange={function (change: boolean): void {
            throw new Error("Function not implemented.");
        } }/>
    );

    let ratingComponent = screen.getByLabelText(/^Champion name/i);
    expect(ratingComponent.innerHTML).toMatch("");


    ratingComponent.innerHTML = "testing";

    expect(ratingComponent.innerHTML).toMatch("testing");

});
