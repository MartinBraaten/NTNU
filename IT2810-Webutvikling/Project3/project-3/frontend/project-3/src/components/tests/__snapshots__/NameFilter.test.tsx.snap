// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1slyxxv"
    role="NameFilter"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-8iuecz-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-15ng2gq-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Champion name
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1sepomn-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-12qs7hp-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-basic"
          type="text"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11frh41-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-cu9mew-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Champion name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
